src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureBlobStorage.cs

```diff
@@ -91,8 +91,8 @@ public async Task ReadStateAsync(string grainType, GrainReference grainId, IGrai
                     grainState.RecordExists = true;
                 }
 
-                var loadedState = this.ConvertFromStorageFormat(contents, grainState.Type);
-                grainState.State = loadedState ?? Activator.CreateInstance(grainState.Type);
+                var loadedState = this.ConvertFromStorageFormat<T>(contents);
+                grainState.State = loadedState ?? Activator.CreateInstance<T>();
```

Comment:
I wonder if we should expose activators out of the serialization library so that custom constructors can be used, with DI, etc.

Reply:
I think it's a good idea, I will try to add that in the next PR